pkg/plugins/discovery_unix.go

```diff
@@ -8,33 +8,22 @@ import (
 	"github.com/docker/docker/pkg/rootless"
 )
 
-func rootlessConfigPluginsPath() string {
-	configHome, err := homedir.GetConfigHome()
-	if err == nil {
-		return filepath.Join(configHome, "docker/plugins")
-	}
-
-	return "/etc/docker/plugins"
-}
-
-func rootlessLibPluginsPath() string {
-	libHome, err := homedir.GetLibHome()
-	if err == nil {
-		return filepath.Join(libHome, "docker/plugins")
-	}
-
-	return "/usr/lib/docker/plugins"
-}
-
-// SpecsPaths returns
-// { "%programdata%\docker\plugins" } on Windows,
-// { "/etc/docker/plugins", "/usr/lib/docker/plugins" } on Unix in non-rootless mode,
-// { "$XDG_CONFIG_HOME/docker/plugins", "$HOME/.local/lib/docker/plugins" } on Unix in rootless mode
-// with fallback to the corresponding path in non-rootless mode if $XDG_CONFIG_HOME or $HOME is not set.
-func SpecsPaths() []string {
+// specsPaths is the non-Windows implementation of [SpecsPaths].
+func specsPaths() []string {
+	var paths []string
 	if rootless.RunningWithRootlessKit() {
-		return []string{rootlessConfigPluginsPath(), rootlessLibPluginsPath()}
+		if configHome, err := homedir.GetConfigHome(); err == nil {
```

Comment:
This change doesn't really seem better to me.
A lot more difficult to read, at least.
Curious why the change?

Reply:
Ah, yes, I can drop this for now. I was going back-and-forth a bit on this one, and I was considering to propose "perhaps we should always prepend the XDG dirs before system-wide paths"

As part of that, I had this; 

```go
// specsPaths is the non-Windows implementation of [SpecsPaths].
func specsPaths() []string {
	var paths []string
	if configHome, err := homedir.GetConfigHome(); err == nil {
		paths = append(paths, filepath.Join(configHome, "docker/plugins"))
	}
	if libHome, err := homedir.GetLibHome(); err == nil {
		paths = append(paths, filepath.Join(libHome, "docker/plugins"))
	}
	return append(paths, "/etc/docker/plugins", "/usr/lib/docker/plugins")
}
```

But then decided to keep that for later, which is how I ended up with the variant here.

I'll revert that part for now, and leave that for a later discussion.